lib/project_types/script/layers/infrastructure/script_project_repository.rb

```diff
@@ -140,40 +138,109 @@ def validate_metadata!(extension_point_type, language)
           end
         end
 
-        class ScriptJsonRepository
+        def script_config_yml_repository
+          @script_config_yml_repository ||= ScriptConfigYmlRepository.new(ctx: ctx)
+        end
+
+        def script_json_repository
+          @script_json_repository ||= ScriptJsonRepository.new(ctx: ctx)
+        end
```

Comment:
Since we're using inheritance, I think we can define a `script_config_repository` method that returns a `ScriptConfigRepository`, whether it's backed by `ScriptJsonRepository` or `ScriptConfigYmlRepository`.

Each of the repositories could define an `active?` method, like Jacob suggested, as a way of determining in priority order which to use.

Reply:
I added this